site/src/main/java/uk/nhs/digital/ps/beans/BaseDocument.java

```diff
@@ -63,4 +64,36 @@ protected RestrictableDate nominalPublicationDateCalendarToRestrictedDate(final
             ? RestrictableDate.restrictedDateFrom(nominalPublicationDate)
             : RestrictableDate.fullDateFrom(nominalPublicationDate);
     }
+
+    /*
+    * Converts selected geographic coverage values:
+    * If England, Wales, Scotland are checked and no others - Great Britain
+    * If England, Wales, Scotland and Northern Ireland are checked and no others - United Kingdom
+    * If England, Wales, Scotland and Northern Ireland, and Republic of Ireland are ticked - British Isles
+    * If any other combination is checked, serve up a list of the checked values
+    */
+    protected String[] geographicCoverageValuesToRegionValue(final String[] cmsValues) {
+
+        if (cmsValues == null) {
+            return null;
+        }
+
+        final String[] greatBritain = new String[] {"England", "Scotland", "Wales"};
+        final String[] unitedKingdom = new String[] {"England", "Scotland", "Wales", "Northern Ireland"};
+        final String[] britishIsles = new String[] {"England", "Scotland", "Wales", "Northern Ireland", "Republic of Ireland"};
+
+        if (cmsValues.length == 3 && containsAllElementsInArray(cmsValues, greatBritain)) {
```

Comment:
Why do we need the length check?

Reply:
Removed check for length, but had to change the array compare function.